darwin/client.py

```diff
@@ -468,52 +414,274 @@ def get_datasets_dir(self, team: Optional[str] = None) -> Optional[str]:
 
         Parameters
         ----------
         datasets_dir: Path
-            Path to set as dataset directory of the team
-        team: str
-            Team to change the directory to
+            Path to set as dataset directory of the team.
+        team: Optional[str]
+            Team slug of the team the dataset will belong to. Defaults to None.
         """
         self.config.put(f"teams/{team or self.default_team}/datasets_dir", datasets_dir)
 
-    def _get_headers(self, team: Optional[str] = None) -> Dict[str, str]:
-        """Get the headers of the API calls to the backend.
+    def confirm_upload(self, dataset_item_id: int, team: Optional[str] = None) -> None:
+        """
+        Confirms that the item was uploaded.
 
         Parameters
         ----------
+        dataset_item_id: int
+            The id of the DatasetItem that was uploaded.
+        team: Optional[str]
+            Team slug of the team the dataset will belong to. Defaults to None.
+        """
+        the_team: Optional[Team] = self.config.get_team(team or self.default_team)
+
+        if not the_team:
+            return None
+
+        team_slug: str = the_team.slug
+
+        self._put_raw(endpoint=f"/dataset_items/{dataset_item_id}/confirm_upload", payload={}, team=team_slug)
+
+    def sign_upload(self, dataset_item_id: int, team: Optional[str] = None) -> Optional[Dict[str, Any]]:
+        """
+        Signs the upload of the given DatasetItem.
+
+        Parameters
+        ----------
+        dataset_item_id: int
+            The id of the DatasetItem that was uploaded.
+        team: Optional[str]
+            Team slug of the team the dataset will belong to. Defaults to None.
 
         Returns
-        -------
-        dict
-        Contains the Content-Type and Authorization token
+        ------
+        Optional[Dict[str, Any]]
+            A dictionary with the signed response, or None if the Team was not found.
         """
-        headers: Dict[str, str] = {"Content-Type": "application/json"}
-        api_key: Optional[str] = None
-        team_config: Optional[Team] = self.config.get_team(team or self.default_team, raise_on_invalid_team=False)
+        the_team: Optional[Team] = self.config.get_team(team or self.default_team)
 
-        if team_config:
-            api_key = team_config.api_key
+        if not the_team:
+            return None
 
-        if api_key and len(api_key) > 0:
-            headers["Authorization"] = f"ApiKey {api_key}"
+        team_slug: str = the_team.slug
 
-        from darwin import __version__
+        response: Dict[str, Any] = cast(
+            Dict[str, Any], self._get(f"/dataset_items/{dataset_item_id}/sign_upload", team=team_slug)
+        )
+        return response
 
-        headers["User-Agent"] = f"darwin-py/{__version__}"
-        return headers
+    def upload_data(
+        self, dataset_slug: str, payload: Dict[str, Any], team: Optional[str] = None
```

Comment:
```suggestion
        self, dataset_slug: str, payload: Dict[str, Any], team_slug: Optional[str] = None
```

Reply:
The documentation was incorrect for this case. This is the Team object, not the slug. Fixed it !